spyder/utils/syntaxhighlighters.py

```diff
@@ -552,6 +559,7 @@ def highlight_block(self, text):
             block_nb = self.currentBlock().blockNumber()
             self.outlineexplorer_data[block_nb] = oedata
             self.outlineexplorer_data['found_cell_separators'] = self.found_cell_separators
+            self.outline_explorer_data_update_timer.start(500)

```

Comment:
Maybe ``1000`` is enough to cut the overhead in half? Every second should be fine, I'd think...

Reply:
It is a one shot timer, so it doesn't really matter. The goal is to avoid calling it multiple times while typing a name (slightly slower ) saving a file (massively slower). I took 500ms as being larger than the time between two keys, but still small so it seems fast. I could even decrease it in my opinion as the file saving scenario is the one I really want to avoid.

Reply:
Okay, sure. 500 ms makes sense, then. Thanks!